### PR TITLE
ULS: Partially styled navigation bar

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #182 

This partially updates the navigation bar to the unified style. 
- It is displayed when large titles are enabled.
- Large titles have been enabled on the new Site Address view, so it will show up there.
- When large titles are disabled, the old nav bar style is used.

Specifically what is done:
- The large title color is set from `largeTitleTextColor` .
- The Back button color is set from `navButtonTextColor`.
- The nav bar background color is set from `navBarBackgroundColor`.
- The bottom line on the nav bar is hidden.

And what is not done:
- The 'Help' button is still white (on the white background, you may not see it, but it's there).
- The app icon is still displayed (it's white, so you may not see it, but it's there).
- The large title font has not been updated.

Can be tested with WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14386

### CODE REVIEW

File: WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift [hunk 1]
```diff
@@ -68,6 +68,10 @@ public struct WordPressAuthenticatorStyle {
 
     public let navBarBadgeColor: UIColor
 
+    public let navBarBackgroundColor: UIColor
+    
+    public let navButtonTextColor: UIColor
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift:73]: These were added to the old Style struct so Auth knows how to style the nav bar when going from a new unified view to an old view.

File: WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift [hunk 2]
```diff
@@ -79,10 +83,10 @@ public struct WordPressAuthenticatorStyle {
...
+    public init(primaryNormalBackgroundColor: UIColor, primaryNormalBorderColor: UIColor?, primaryHighlightBackgroundColor: UIColor, primaryHighlightBorderColor: UIColor?, secondaryNormalBackgroundColor: UIColor, secondaryNormalBorderColor: UIColor, secondaryHighlightBackgroundColor: UIColor, secondaryHighlightBorderColor: UIColor, disabledBackgroundColor: UIColor, disabledBorderColor: UIColor, primaryTitleColor: UIColor, secondaryTitleColor: UIColor, disabledTitleColor: UIColor, textButtonColor: UIColor, textButtonHighlightColor: UIColor, instructionColor: UIColor, subheadlineColor: UIColor, placeholderColor: UIColor, viewControllerBackgroundColor: UIColor, textFieldBackgroundColor: UIColor, buttonViewBackgroundColor: UIColor? = nil, navBarImage: UIImage, navBarBadgeColor: UIColor, navBarBackgroundColor: UIColor, navButtonTextColor: UIColor = .white, prologueBackgroundColor: UIColor = WPStyleGuide.wordPressBlue(), prologueTitleColor: UIColor = .white, statusBarStyle: UIStatusBarStyle = .lightContent) {
```
Comment (Gio2018) [inline WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift:89]: Minor thing, but maybe put each argument in a new line could increase readability?

File: WordPressAuthenticator/Unified Auth/View Related/SiteAddressViewController.swift [hunk 3]
```diff
@@ -22,6 +22,9 @@ final class SiteAddressViewController: LoginViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        navigationItem.title = WordPressAuthenticator.shared.displayStrings.logInTitle
+        setLargeTitleDisplayMode(.always)
+
```
Comment (mindgraffiti) [inline WordPressAuthenticator/Unified Auth/View Related/SiteAddressViewController.swift:27]: Yay thank you!

File: WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift (hunk 2 above)
Comment (ScoutHarris) [reply WordPressAuthenticator/Authenticator/WordPressAuthenticatorStyles.swift:89]: Ya know, I was annoyed by this but didn't think to change it. 🤦 Will do!
